ext/web/13_message_port.js

```diff
@@ -145,6 +146,17 @@ class MessagePort extends EventTarget {
     op_message_port_post_message(this[_id], data);
   }
 
+  /**
+   * This method is only provided for `receiveMessageOnPort`
+   * in `node:worker_threads`
+   * @returns {object | undefined}
+   */
+  receiveMessage() {
+    const data = op_message_port_recv_message_sync(this[_id]);
+    if (data === null) return undefined;
+    return { message: deserializeJsMessageData(data)[0] };
+  }
```

Comment:
`MessagePort` is a Web standard API and it doesn't have this method. We shouldn't add it. Instead use the body of this method in `receiveMessageOnPort` directly.

Reply:
Yes -- I agree with you. I was trying to implement it in this way first, but I couldn't figure out any solution to get access to the private RID properties of the `MessagePort` class from `workers_threads`, where it should belong. 

That's why I have chosen at the end this compromise, which looks at least more acceptable to me than leaking internal class data.
 
And this function is so close related and interwoven with all the `MessageChannel` and `MessagePort` internal infastructure, that it somehow fits very well there. 

Another solution could be perhaps a derived class -- similar to the `nodeEventTarget` construct -- to isolate the different specification contexts resp. their overlapping margins, but that's IMHO also not a really attractive solution. 

Well -- I can change it in any way you like as long as it is technically possible. At the end it's just a matter of convincing arguments.

Reply:
ps: Renaming and using a leading underscore would be another simple solution to clearly signify the proprietary nature of this method.

Reply:
You can export `_id` from `13_message_port.js` and then import it in `worker_threads.ts` which will enable you to move all that `receiveMessage` logic into` receiveMessageOnPort` directly.

> ps: Renaming and using a leading underscore would be another simple solution to clearly signify the proprietary nature of this method.

Once people find this API exist they will eventually start to rely on it and it will make it very hard to remove it, let's just make sure that all this fix is completely internal and doesn't expose any new methods and fields besides `worker_threads::receiveMessageOnPort` :)

Reply:
You are right! -- I share your concerns.

 I'll try to create a solution based on your suggestion (exporting the `_id` symbol) over the weekend.